Client.java

```diff
@@ -1,80 +1,56 @@
+
+//   private static final int MAX_MOVES = 1000;
+
+   private static void nextMoveFor(Player player)
+   {
+      int spin = Board.spin();
+      System.out.println(player.getName() + " spins: " + spin);
+      player.setSpace(Board.getNewLocation(player.getSpace(), spin));
+      System.out.println(player + "\n\n");
+   }
+   public static void main(String[] args)
+   {
       Scanner input = new Scanner(System.in);
       System.out.println("How many players (2 - 4)?");
-      int plrNum = input.nextInt();
-      
-      Player p1 = new Player("Bob");
-      Player p2 = new Player("Joe");
-      Player p3 = new Player("Bill");
-      Player p4 = new Player("Kevin");
-      
-      Board b = new Board();
-      
-      while(p1.getSpace() < 100 && p2.getSpace() < 100 && p3.getSpace() < 100 && p4.getSpace() < 100){
-      
-         if(plrNum >= 1){
-            int spin = b.spin();
-            System.out.println(p1.getName() + " spins: " + spin);
-            int newLoc = b.getNewLocation(p1.getSpace(), spin);
-            p1.setSpace(newLoc);
-            System.out.println(p1.getName() + " is on space " + p1.getSpace());  
-            System.out.println("");
-            System.out.println("");
-         }
-         
-         if(p1.getSpace() == 100){
-            System.out.println(p1.getName() + " wins!");
-            break;
-         }
-         
-         if(plrNum >= 2){
-            int spin = b.spin();
-            System.out.println(p2.getName() + " spins: " + spin);
-            int newLoc = b.getNewLocation(p2.getSpace(), spin);
-            p2.setSpace(newLoc);
-            System.out.println(p2.getName() + " is on space " + p2.getSpace());  
-            System.out.println("");
-            System.out.println("");
-         }
-         
-         if(p2.getSpace() == 100){
-            System.out.println(p2.getName() + " wins!");
-            break;
-         }
-         
-         if(plrNum >= 3){
-            int spin = b.spin();
-            System.out.println(p3.getName() + " spins: " + spin);
-            int newLoc = b.getNewLocation(p3.getSpace(), spin);
-            p3.setSpace(newLoc);
-            System.out.println(p3.getName() + " is on space " + p3.getSpace());  
-            System.out.println("");
-            System.out.println("");
-         }
-         
-         if(p3.getSpace() == 100){
-            System.out.println(p3.getName() + " wins!");
-            break;
+      int gamePlayersNumber = input.nextInt();
+
+//      if (gamePlayersNumber>4 || gamePlayersNumber<2)
+//      {
+//         System.out.println("Number of players must be 2, 3 or 4.");
+//         System.out.println("Please, enter a valid number next time you start the game simulator.");
+//         return;
+//      }
+
+      Player[] players = new Player[gamePlayersNumber];
+      for (int i=0; i<gamePlayersNumber; i++)
+      { players[i] =  new Player(PLAYER_NAMES[i]); }
+
+      boolean gameEnded = false;

```

Comment:
i dunno if this would b called a 'trick' by your teacher - adding an extra indicator 2 control when the game loop should end (vs your original implementation or checking all player states)...